parser-utils/src/lib.rs

```diff
@@ -8,6 +8,14 @@
 // CONDITIONS OF ANY KIND, either express or implied. See the License for the
 // specific language governing permissions and limitations under the License.
 
+extern crate combine;
+
+use combine::ParseResult;
+use combine::combinator::{Expected, FnParser};
+
+// Nothing about this is specific to the type of parser.
+pub type ResultParser<O, I> = Expected<FnParser<I, fn(I) -> ParseResult<O, I>>>;
```

Comment:
Some kind of explanation about what this is intended for would still be nice (at least for people like me).